### PR TITLE
Update the KeyVault for Test pipelines

### PR DESCRIPTION
**Pull request by yophilav**
1. Reinstitute LongHaul IotHub for both SingleNode & NestedEdge for testing
2. Consolidate KeyVault variables so the same type of tests are sharing the same IoTHub.

Remark: With the keyVault variable consolidation (2), a developer can now inject his own keyVault into the pipeline to have any of the test pipeline connect upstream to his/her own IoTHub. The two required "secrets" in the developer's keyVaults are `IotHub-ConnStr` (IoTHub Connection String) and `IotHub-EventHubConnStr` (IoTHub's built-in EventHub Endpoint Connection String)

### CODE REVIEW

**Comment by yophilav** [comment]
For the reviewers, the following test runs are proofs the changes work for all the mentioned pipelines. 
[SingleNode LongHaul](https://msazure.visualstudio.com/One/_build/results?buildId=41859570&view=results)
[Nested LongHaul](https://msazure.visualstudio.com/One/_build/results?buildId=41864114&view=results)
[Nested E2E](https://msazure.visualstudio.com/One/_build/results?buildId=41861826&view=results)
[ISAl95](https://msazure.visualstudio.com/One/_build/results?buildId=41861823&view=results)
[SingleNode Connectivity](https://msazure.visualstudio.com/One/_build/results?buildId=41863790&view=results)
[Nested Connectivity](https://msazure.visualstudio.com/One/_build/results?buildId=41863784&view=results)

File: builds/e2e/longhaul.yaml [hunk 1]
```diff
@@ -285,11 +317,27 @@ jobs:
...
-        displayName: 'Azure Key Vault'
+        displayName: 'Common Test Azure Key Vault'
+        inputs:
+          azureSubscription: $(azure.subscription)
+          KeyVaultName: 'edgebuildkv'
+          SecretsFilter: >- 
+            edgebuilds-azurecr-io-username,
+            edgebuilds-azurecr-io-pwd,
+            StorageAccountMasterKeyStress,
+            SnitchLongHaulAlertUrl,
```
Comment (and-rewsmith) [inline builds/e2e/longhaul.yaml:328]: Ideally we would remove this, but I think it is better to have a separate scoped PR for removing the old longhaul logic (containing lingering refs to snitcher and analyzer).

File: builds/e2e/connectivity.yaml [hunk 2]
```diff
@@ -106,10 +106,24 @@ jobs:
       - task: AzureKeyVault@1
         condition: eq(variables['run.flag'], 1)
         displayName: 'Azure Key Vault'
```
Comment (and-rewsmith) [inline builds/e2e/connectivity.yaml:108]: Can we indicate which key vault this is coming from like we do in other places?

Comment (and-rewsmith) [reply builds/e2e/connectivity.yaml:108]: It might be worth updating the other refs in the repo too

**Review by and-rewsmith — APPROVED** [review]

**Review by huguesBouvier — APPROVED** [review]

File: builds/e2e/templates/nested-get-secrets.yaml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 steps:
   - task: AzureKeyVault@1
-    displayName: 'Common Azure Key Vault'
+    displayName: 'Azure Key Vault: EdgeBuildkv'
```
Comment (and-rewsmith) [inline builds/e2e/templates/nested-get-secrets.yaml:3]: super nit: the casing here is not how it is for the portal resource

**Review by and-rewsmith — APPROVED** [review]

**Review by dylanbronson — APPROVED** [review]
LGTM nice change

**Pull request merged**